internal/libyaml/reader.go

```diff
@@ -23,15 +23,25 @@
 package libyaml
 
 import (
+	"errors"
+	"fmt"
 	"io"
 )
 
 // Set the reader error and return 0.
 func (parser *Parser) setReaderError(problem string, offset int, value int) bool {
-	parser.ErrorType = READER_ERROR
-	parser.Problem = problem
-	parser.ProblemOffset = offset
-	parser.ProblemValue = value
+	return parser.directlySetReaderError(
+		errors.New(problem),
+		offset, value,
+	)
+}
+
+func (parser *Parser) directlySetReaderError(err error, offset int, value int) bool {
+	parser.Error = &ReaderError{
+		Offset: offset,
+		Value:  value,
+		Err:    err,
+	}
 	return false
 }
```

Comment:
I feel like you are using this only twice. And all if them in this file

I wouldn't add this method and directly return ReaderError as you do in this method.

Two reasons for this:
- we can live with the duplication cost
- the arguments of this method are unclear from the caller

See https://github.com/yaml/go-yaml/pull/177#discussion_r2500822650